src/lib.rs

```diff
@@ -337,15 +338,15 @@ impl fuse::Filesystem for SandboxFS {
 }
 
 /// Mounts a new sandboxfs instance on the given `mount_point` and maps all `mappings` within it.
-pub fn mount(mount_point: &Path, mappings: &[Mapping]) -> io::Result<()> {
+pub fn mount(mount_point: &Path, mappings: &[Mapping]) -> Result<(), Error> {
     let options = ["-o", "ro", "-o", "fsname=sandboxfs"]
         .iter()
         .map(|o| o.as_ref())
         .collect::<Vec<&OsStr>>();
     let fs = SandboxFS::new(mappings)?;
     info!("Mounting file system onto {:?}", mount_point);
-    fuse::mount(fs, &mount_point, &options)
-        .map_err(|e| io::Error::new(e.kind(), format!("mount on {:?} failed: {}", mount_point, e)))
+    fuse::mount(fs, &mount_point, &options).context(format!("mount on {:?} failed", mount_point))?;
```

Comment:
FYI `context(...)` hides the underlying error.

Reply:
Resolving per the other comment thread on flattening causes, as this is directly derived from that.